File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.c
#include <stdio.h>
#define MAX 10001

int main(){
    int i, sum, idx;
    int arr[MAX] = {1,0 }; //셀프 넘버 = 0 
    
    for(i=1; i<=MAX; i++){
        idx = sum = i;
        while(idx){
            sum += idx%10;
            idx /= 10;
        }
        if(sum<=MAX) arr[sum] = 1;
    }
    
    for(i=0; i<=MAX; i++)
        if(arr[i]==0)
            printf("%d\n", i);
            
    return 0;
}

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.java
class Main {
  static int MAX = 10001;  
  static boolean[] check = new boolean[MAX]; //셀프넘버 = 0

  static void f(int i){
    int sum = i;
    while(i>0){
      sum += i%10;
      i /= 10;
    }
    if(sum<MAX) check[sum] = true;
  }
  
  public static void main(String[] args) {
    for(int i=1; i<MAX; i++)
      f(i);
    
    for(int i=1; i<MAX; i++)
      if(!check[i]) System.out.println(i);
    
    
  }
}